src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityNpc.kt

```diff
@@ -5,30 +5,58 @@ import at.hannibal2.skyhanni.events.InventoryCloseEvent
 import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
 import at.hannibal2.skyhanni.events.InventoryUpdatedEvent
 import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.features.fame.ReminderUtils
+import at.hannibal2.skyhanni.features.inventory.chocolatefactory.ChocolateFactoryAPI
+import at.hannibal2.skyhanni.utils.ChatUtils
 import at.hannibal2.skyhanni.utils.InventoryUtils
 import at.hannibal2.skyhanni.utils.ItemUtils.getLore
 import at.hannibal2.skyhanni.utils.LorenzColor
 import at.hannibal2.skyhanni.utils.LorenzUtils
 import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockTime
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
 
 object HoppityNpc {
 
     private val config get() = HoppityEggsManager.config
 
+    private var lastReminderSent = SimpleTimeMark.farPast()
+    private var hoppityYearOpened
+        get() = ChocolateFactoryAPI.profileStorage?.hoppityShopYearOpened ?: -1
+        set(value) {
+            ChocolateFactoryAPI.profileStorage?.hoppityShopYearOpened = value
+        }
+
     private var slotsToHighlight = mutableSetOf<Int>()
     private var inShop = false
 
     @SubscribeEvent
     fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
-        if (!isEnabled()) return
         if (event.inventoryName != "Hoppity") return
+        hoppityYearOpened = SkyBlockTime.now().year
```

Comment:
out of scope of this pr, but maybe we could add an annoying chat message that tells you how many years you have skipped/ the last year you have opened the shop before. that way we verbally punish non active users in a funny and non harmful way :)